### PR TITLE
XW-2747 | WikiaRSS fixes

### PR DESCRIPTION
**Pull request by kvas-damian**
https://wikia-inc.atlassian.net/browse/XW-2747

Ping @Wikia/x-wing @Grunny

### CODE REVIEW

File: extensions/wikia/WikiaRSS/WikiaRssExternalController.class.php [hunk 1]
```diff
@@ -127,6 +133,12 @@ private function getFullItemList($items, $options) {
 			if( $item['description'] ) {
 				$text = trim(mb_convert_encoding($item['description'],$outputEncoding,$charset));
 				$text = str_replace( array("\r", "\n", "\t", '<br>'), ' ', $text );
+				$text = strip_tags( $text, '<img><a><br>' );
```
Comment (rogatty) [inline extensions/wikia/WikiaRSS/WikiaRssExternalController.class.php:136]: Why do we strip tags twice? First here and then `Sanitizer::removeHTMLtags()`. Is it *to be sure*?

File: extensions/wikia/WikiaRSS/WikiaRssExternalController.class.php [hunk 2]
```diff
@@ -127,6 +133,12 @@ private function getFullItemList($items, $options) {
...
+
+				$globalStateWrppaer = new \Wikia\Util\GlobalStateWrapper( [ 'wgAllowImageTag' => true ] );
```
Comment (rogatty) [inline extensions/wikia/WikiaRSS/WikiaRssExternalController.class.php]: Typo in the var name.

File: extensions/wikia/WikiaRSS/WikiaRssExternalController.class.php (hunk 1 above)
Comment (kvas-damian) [reply extensions/wikia/WikiaRSS/WikiaRssExternalController.class.php:136]: `removeHTMLtags()` is more complex method. By defaults it strips tags that are not whitelisted. The same rule is applied to attributes. Other tags are kept in place. Unfortunately this method can't handle more complex HTML (or not fully valid HTML) that's why we limit allowed tags using `strip_tags()`.

Comment (rogatty) [reply extensions/wikia/WikiaRSS/WikiaRssExternalController.class.php:136]: Interesting, sounds like `removeHTMLtags()` should call `strip_tags()` internally. Not in this PR, obviously :slightly_smiling_face:

Comment (kvas-damian) [reply extensions/wikia/WikiaRSS/WikiaRssExternalController.class.php:136]: Name for `removeHTMLtags()` is very misleading. In fact it's trying to modify HTML in a way that it's safe.

**Review by rogatty — APPROVED** [review]

**Pull request merged**